File: screens/OnboardingScreen.js
import React from 'react';
import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
  ScrollView,
} from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import { StatusBar } from 'expo-status-bar';
import { LinearGradient } from 'expo-linear-gradient';

import { useTheme } from '../src/AppTheme';
import Blobs from '../src/components/Blobs';
import PressScale from '../src/components/PressScale';
import { AppLanguages } from '../src/i18n';
import { Colors, Spacing, Shadow } from '../src/theme';

export default function OnboardingScreen({ goNext }) {
  const { accent, language, setLanguage, t } = useTheme();

  const row1 = AppLanguages.slice(0, 3);
  const row2 = AppLanguages.slice(3);

  return (
    <SafeAreaView style={styles.root} edges={['top', 'bottom']}>
      <StatusBar style="dark" />
      <Blobs accent={accent} opacity={1.0} />

      {/* Center content */}
      <View style={styles.centerContent}>
        <LinearGradient
          colors={[Colors.shadowGrey, accent]}
          start={{ x: 0, y: 0 }}
          end={{ x: 1, y: 1 }}
          style={styles.logoBox}
        >
          <Text style={styles.logoEmoji}>⚡</Text>
        </LinearGradient>

        <Text style={styles.appTitle}>GrayMan</Text>

        <Text style={styles.subtitle}>
          {t(
            'Digital identity for skilled workers',
            'कुशल कामगारों की डिजिटल पहचान',
            {
              mr: 'कुशल कामगारांची डिजिटल ओळख',
              te: 'నైపుణ్య కార్మికుల డిజిటల్ గుర్తింపు',
              ta: 'திறன் தொழிலாளர்களின் டிஜிட்டல் அடையாளம்',
              kn: 'ಕೌಶಲ್ಯ ಕಾರ್ಮಿಕರ ಡಿಜಿಟಲ್ ಗುರುತು',
            },
          )}
        </Text>

        <Text style={[styles.devanagariTag, { color: accent }]}>काम · सेतु</Text>
      </View>

      {/* Bottom CTA section */}
      <View style={styles.bottomSection}>
        <PressScale onPress={goNext} style={[styles.primaryBtn, { backgroundColor: accent, shadowColor: accent }]}>
          <Text style={styles.primaryBtnText}>
            {t('Get Started', 'शुरू करें')} →
          </Text>
        </PressScale>

        <PressScale onPress={goNext} style={styles.secondaryBtn}>
          <Text style={styles.secondaryBtnText}>{t('Sign In', 'साइन इन करें')}</Text>
        </PressScale>

        <Text style={styles.termsText}>
          {t(
            'By continuing you agree to our Terms & Privacy',
            'जारी रखकर आप हमारी शर्तों और गोपनीयता से सहमत हैं',
          )}
        </Text>
      </View>

      {/* Language pill grid — pinned above safe-area bottom */}
      <View style={styles.langGrid}>
        <View style={styles.langRow}>
          {row1.map(lang => {
            const selected = language === lang.key;
            return (
              <TouchableOpacity
                key={lang.key}
                onPress={() => setLanguage(lang.key)}
                style={[
                  styles.langPill,
                  selected
                    ? { backgroundColor: accent }
                    : { backgroundColor: Colors.soft },
                ]}
                activeOpacity={0.75}
              >
                <Text
                  style={[
                    styles.langPillText,
                    { color: selected ? '#fff' : Colors.shadowGrey },
                  ]}
                >
                  {lang.shortCode}
                </Text>
              </TouchableOpacity>
            );
          })}
        </View>
        <View style={[styles.langRow, { marginTop: 6 }]}>
          {row2.map(lang => {
            const selected = language === lang.key;
            return (
              <TouchableOpacity
                key={lang.key}
                onPress={() => setLanguage(lang.key)}
                style={[
                  styles.langPill,
                  selected
                    ? { backgroundColor: accent }
                    : { backgroundColor: Colors.soft },
                ]}
                activeOpacity={0.75}
              >
                <Text
                  style={[
                    styles.langPillText,
                    { color: selected ? '#fff' : Colors.shadowGrey },
                  ]}
                >
                  {lang.shortCode}
                </Text>
              </TouchableOpacity>
            );
          })}
        </View>
      </View>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  root: {
    flex: 1,
    backgroundColor: Colors.canvas,
  },
  centerContent: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    paddingHorizontal: 24,
  },
  logoBox: {
    width: 82,
    height: 82,
    borderRadius: 26,
    alignItems: 'center',
    justifyContent: 'center',
    marginBottom: 20,
  },
  logoEmoji: {
    fontSize: 34,
    color: '#fff',
  },
  appTitle: {
    fontSize: 44,
    fontWeight: '800',
    color: Colors.shadowGrey,
    letterSpacing: -1.5,
    marginBottom: 10,
  },
  subtitle: {
    fontSize: 16,
    color: Colors.mutedText,
    textAlign: 'center',
    maxWidth: 260,
    lineHeight: 22,
    marginBottom: 14,
  },
  devanagariTag: {
    fontSize: 13,
    fontWeight: '700',
    letterSpacing: 0.6,
  },

  // Bottom section
  bottomSection: {
    paddingHorizontal: 24,
    paddingBottom: 100,
    alignItems: 'center',
  },
  primaryBtn: {
    width: '100%',
    paddingVertical: 17,
    borderRadius: 16,
    alignItems: 'center',
    marginBottom: 12,
    ...Shadow.button,
  },
  primaryBtnText: {
    color: '#fff',
    fontSize: 16,
    fontWeight: '700',
  },
  secondaryBtn: {
    width: '100%',
    paddingVertical: 17,
    borderRadius: 16,
    alignItems: 'center',
    borderWidth: 1.5,
    borderColor: 'rgba(39,41,50,0.18)',
    marginBottom: 16,
  },
  secondaryBtnText: {
    color: Colors.shadowGrey,
    fontSize: 16,
    fontWeight: '600',
  },
  termsText: {
    fontSize: 12,
    color: Colors.dimText,
    textAlign: 'center',
    lineHeight: 17,
  },

  // Language pills
  langGrid: {
    position: 'absolute',
    bottom: 32,
    left: 0,
    right: 0,
    alignItems: 'center',
  },
  langRow: {
    flexDirection: 'row',
    gap: 8,
  },
  langPill: {
    width: 52,
    height: 32,
    borderRadius: 10,
    alignItems: 'center',
    justifyContent: 'center',
  },
  langPillText: {
    fontSize: 13,
    fontWeight: '600',
  },
});

File: flutter_app/lib/features/onboarding/onboarding_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import '../../core/design_system/app_colors.dart';
import '../../core/design_system/app_theme.dart';
import '../../core/design_system/blobs.dart';
import '../../core/design_system/press_scale.dart';

class OnboardingScreen extends StatelessWidget {
  const OnboardingScreen({super.key, required this.onNext});
  final VoidCallback onNext;

  @override
  Widget build(BuildContext context) {
    final theme = context.watch<GrayManTheme>();
    return Scaffold(
      backgroundColor: AppColors.canvas,
      body: Stack(
        children: [
          Blobs(accent: theme.accent, opacity: 1),
          SafeArea(
            child: Column(
              children: [
                const Spacer(),
                Container(
                  width: 82,
                  height: 82,
                  decoration: BoxDecoration(
                    borderRadius: BorderRadius.circular(26),
                    gradient: LinearGradient(colors: [AppColors.shadowGrey, theme.accent], begin: Alignment.topLeft, end: Alignment.bottomRight),
                    boxShadow: [BoxShadow(color: theme.accent.withValues(alpha: 0.42), blurRadius: 14, offset: const Offset(0, 8))],
                  ),
                  child: const Icon(Icons.bolt, color: Colors.white, size: 34),
                ),
                const SizedBox(height: 28),
                Text('sthapna.ai', style: TextStyle(fontSize: 44, fontWeight: FontWeight.w900, color: AppColors.shadowGrey, letterSpacing: -1.5)),
                const SizedBox(height: 8),
                Text(
                  theme.t('Digital identity for skilled workers', 'कुशल कामगारों की डिजिटल पहचान'),
                  textAlign: TextAlign.center,
                  style: const TextStyle(fontSize: 16, color: AppColors.mutedText),
                ),
                const SizedBox(height: 6),
                Text('काम · सेतु', style: TextStyle(fontSize: 13, fontWeight: FontWeight.bold, color: theme.accent, letterSpacing: 0.6)),
                const Spacer(),
                Padding(
                  padding: const EdgeInsets.fromLTRB(24, 0, 24, 100),
                  child: Column(
                    children: [
                      PressScaleButton(
                        onPressed: onNext,
                        child: Container(
                          width: double.infinity,
                          padding: const EdgeInsets.symmetric(vertical: 17),
                          decoration: BoxDecoration(color: theme.accent, borderRadius: BorderRadius.circular(16), boxShadow: [BoxShadow(color: theme.accent.withValues(alpha: 0.36), blurRadius: 12, offset: const Offset(0, 6))]),
                          child: Text(theme.t('Get Started', 'शुरू करें'), textAlign: TextAlign.center, style: const TextStyle(color: Colors.white, fontSize: 16, fontWeight: FontWeight.bold)),
                        ),
                      ),
                      const SizedBox(height: 12),
                      PressScaleButton(
                        onPressed: onNext,
                        scale: 0.98,
                        child: Container(
                          width: double.infinity,
                          padding: const EdgeInsets.symmetric(vertical: 16),
                          decoration: BoxDecoration(borderRadius: BorderRadius.circular(16), border: Border.all(color: AppColors.shadowGrey.withValues(alpha: 0.18))),
                          child: Text(theme.t('Sign In', 'साइन इन'), textAlign: TextAlign.center, style: const TextStyle(color: AppColors.shadowGrey, fontSize: 15, fontWeight: FontWeight.w600)),
                        ),
                      ),
                      const SizedBox(height: 12),
                      Text(theme.t('By continuing you agree to our Terms & Privacy', 'जारी रखकर आप हमारी शर्तों और गोपनीयता से सहमत हैं'), textAlign: TextAlign.center, style: const TextStyle(fontSize: 12, color: AppColors.dimText)),
                    ],
                  ),
                ),
              ],
            ),
          ),
        ],
      ),
    );
  }
}
